src/main/kotlin/no/nav/tag/innsynAareg/service/InnsynService.kt

```diff
@@ -99,45 +99,43 @@ class InnsynService(
             overOrdnetEnhetOrgnr
         }
 
-        val arbeidsforhold = aaregClient.hentArbeidsforhold(bedriftsnr, opplysningspliktigorgnr, idPortenToken)
-
-        if (arbeidsforhold is ArbeidsforholdFunnet) {
-            navneoppslagService.settNavn(arbeidsforhold.oversiktOverArbeidsForhold)
-            settYrkeskodebetydningPaAlleArbeidsforhold(arbeidsforhold.oversiktOverArbeidsForhold)
+        return aaregClient.hentArbeidsforhold(bedriftsnr, opplysningspliktigorgnr, idPortenToken).apply {
+            if (this is ArbeidsforholdFunnet) {
+                navneoppslagService.settNavn(oversiktOverArbeidsForhold)
+                settYrkeskodebetydningPaAlleArbeidsforhold(oversiktOverArbeidsForhold)
+            }
         }
-
-        return arbeidsforhold
     }
 
-
     fun hentTidligereArbeidsforhold(
         bedriftsnr: String,
         overOrdnetEnhetOrgnr: String,
         idPortenToken: String,
         fnr: String
     ): ArbeidsforholdOppslagResultat {
-
-        var oversiktOverArbeidsforhold = aaregClient.hentArbeidsforhold(
+        val oversiktOverArbeidsforhold = aaregClient.hentArbeidsforhold(
             bedriftsnr,
             overOrdnetEnhetOrgnr,
             idPortenToken
         )
 
-        if (oversiktOverArbeidsforhold !is ArbeidsforholdFunnet) {
-            oversiktOverArbeidsforhold = finnOpplysningspliktigOgHentArbeidsforhold(
-                bedriftsnr,
-                overOrdnetEnhetOrgnr,
-                idPortenToken,
-                fnr
-            )
-        }
-
-        if (oversiktOverArbeidsforhold is ArbeidsforholdFunnet) {
-            navneoppslagService.settNavn(oversiktOverArbeidsforhold.oversiktOverArbeidsForhold)
-            settYrkeskodebetydningPaAlleArbeidsforhold(oversiktOverArbeidsforhold.oversiktOverArbeidsForhold)
+        return oversiktOverArbeidsforhold.let {
```

Comment:
denne koden gjør ikke det samme som den forrige. Hvis man får `IngenRettigheter`, så forsøker den og hente på opplysningspliktig, og utfører navneoppslag og yrkeskodeoppslag på resultat derfra (hvis det ble funnet).

Reply:
Kanskje noe a la
```
forhold = when (val x = hent1()) {
   is Funnet -> x
   is IkkeFunnet -> hent2()
}

return when (forhold) {
 IkkeFunnet -> forhold
 Funnet -> leggPåNavnOsv()
}
```

Reply:
det så jeg ikke

Reply:
reverter akkurat denne delen. var ikke så lett å se, hadde kanskje vært bedre med to egne paths i stedet for å overskrive oversiktOverArbeidsforhold. kan se på denne en lukedag kanskje